src/Symfony/Component/Security/Core/Authorization/SwitchUserAuthorizationChecker.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization;
+
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Role\SwitchUserRole;
+
+/**
+ * SwitchUserAuthorizationChecker gives access to the token representing the impersonated user.
+ *
+ * @author Fabien Papet <fabien.papet@gmail.com>
+ */
+class SwitchUserAuthorizationChecker implements SwitchUserAuthorizationCheckerInterface
+{
+    /**
+     * @var AccessDecisionManagerInterface
+     */
+    private $accessDecisionManager;
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    public function __construct(AccessDecisionManagerInterface $accessDecisionManager, TokenStorageInterface $tokenStorage)
+    {
+        $this->accessDecisionManager = $accessDecisionManager;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function isGranted($attributes, $subject = null)
+    {
+        $token = $this->getOriginalToken($this->tokenStorage->getToken());
+
+        if (false !== $token) {
+            return $this->accessDecisionManager->decide($token, $attributes, $subject);
+        }
+
+        // There is no impersonator so do not grant access
+        return false;
+    }
+
+    /**
+     * Gets the original Token from a switched one.
+     *
+     * @param TokenInterface $token
+     *
+     * @return TokenInterface|false The original TokenInterface instance, false if the current TokenInterface is not switched
+     */
+    private function getOriginalToken(TokenInterface $token)
```

Comment:
As of 4.3 you can simply check that the token is an instance of `SwitchUserToken` and remove this method :). See https://github.com/symfony/symfony/pull/22048

Reply:
Code is from october2018 it didn't exist where I did this. I'll edit this evening